### ISSUE TITLE
yaml linting stops processing in tasks that follow a task tagged with `skip_ansible_lint`

### ISSUE THREAD

**Issue by gardar** [ansible/ansible-lint#3139]
##### Summary

When you tag a task with `skip_ansible_lint` it will stop all yaml linting (yamllint) in the tasks that follow. 

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.14.1.dev0 using ansible 2.14.3
You are using a pre-release version of ansible-lint.

yamllint --version
yamllint 1.29.0
```

- ansible installation method: OS package
- ansible-lint installation method: OS package

##### STEPS TO REPRODUCE

Put in a task that violates a yaml lint rule (line-length) for example.
Then have a task that's tagged with `skip_ansible_lint` 
Then have another task below that also violates the yaml lint.

Running `yammlint` will produce two errors, but running `ansible-lint` will only produce one.

If the `skip_ansible_lint` task is removed, then `ansible-lint` will produce two errors.

Adding other tasks that produce linting errors (not yaml ones) (at least jinja - haven't tried all types) before and after the `skip_ansible_lint` task works as expected, they will produce errors before and after.

```yaml (paste below)
---
- name: Very long line before tag
  ansible.builtin.debug:
    msg: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi pharetra placerat nisi eget efficitur. Nulla sollicitudin, lectus cursus luctus rhoncus, massa tortor faucibus ligula, nec aliquet nulla."

- name: Non yaml ansible-lint violation before tag
  ansible.builtin.debug:
    msg: "{{ ansible-lint will not like this }}"

- name: Set a tag
  ansible.builtin.debug:
    msg: "The tag from this tag should not affect other tags"
  tags:
    - skip_ansible_lint

- name: Very long line after tag
  ansible.builtin.debug:
    msg: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi pharetra placerat nisi eget efficitur. Nulla sollicitudin, lectus cursus luctus rhoncus, massa tortor faucibus ligula, nec aliquet nulla."

- name: Non yaml ansible-lint violation after tag
  ansible.builtin.debug:
    msg: "{{ ansible-lint will not like this }}"
```

##### Desired Behavior
To get yaml lint errors after tasks tagged with `skip_ansible_lint` such as:

```console
WARNING  Listing 4 violation(s) that are fatal
yaml[line-length]: Line too long (211 > 160 characters)
tasks/test.yml:4

jinja[invalid]: template error while templating string: expected token 'end of print statement', got 'will'. String: {{ ansible-lint will not like this }}. expected token 'end of print statement', got 'will'
tasks/test.yml:8 Task/Handler: Non yaml ansible-lint violation before tag

yaml[line-length]: Line too long (211 > 160 characters)
tasks/test.yml:18

jinja[invalid]: template error while templating string: expected token 'end of print statement', got 'will'. String: {{ ansible-lint will not like this }}. expected token 'end of print statement', got 'will'
tasks/test.yml:22 Task/Handler: Non yaml ansible-lint violation after tag

Read documentation for instructions on how to ignore specific rule violations.

                Rule Violation Summary
 count tag               profile rule associated tags
     2 jinja[invalid]    basic   formatting
     2 yaml[line-length] basic   formatting, yaml

Failed after min profile: 4 failure(s), 0 warning(s) on 1 files.
```

##### Actual Behavior

Yaml linting error is ignored silently after the tag.

```console
WARNING  Listing 3 violation(s) that are fatal
yaml[line-length]: Line too long (211 > 160 characters)
tasks/test.yml:4

jinja[invalid]: template error while templating string: expected token 'end of print statement', got 'will'. String: {{ ansible-lint will not like this }}. expected token 'end of print statement', got 'will'
tasks/test.yml:8 Task/Handler: Non yaml ansible-lint violation before tag

jinja[invalid]: template error while templating string: expected token 'end of print statement', got 'will'. String: {{ ansible-lint will not like this }}. expected token 'end of print statement', got 'will'
tasks/test.yml:22 Task/Handler: Non yaml ansible-lint violation after tag

Read documentation for instructions on how to ignore specific rule violations.

                Rule Violation Summary
 count tag               profile rule associated tags
     2 jinja[invalid]    basic   formatting
     1 yaml[line-length] basic   formatting, yaml

Failed after min profile: 3 failure(s), 0 warning(s) on 1 files.
```

**Comment by ssbarnea** [comment]
I am unable to reproduce the current bug with a minimized sample:

```
---
- name: Foo
  hosts: localhost
  tasks:
    - name: "1"
      debug:
        msg: foo

    - name: "2"
      debug:
        msg: foo
      tags:
        - skip_ansible_lint

    - name: "3"
      debug:
        msg: foo
```

![](https://sbarnea.com/ss/Screen-Shot-2023-03-06-17-16-39.01.png)

As you can see adding the tag silences this particular task. I suspect that the bug trigger might be little bit different than it was originally described.

**Comment by gardar** [comment]
What error do you get with your playbook? I get `fqcn[action-core]` which is not a part of the yaml lint, the yaml lint is what's being skipped.
Could you please test with the playbook I provided?

**Comment by ssbarnea** [comment]
@gardar I was able to reproduce it and I have a fix in progress, is just that you might not find the fix ideal.

**Comment by gardar** [comment]
That's fine by me, I usually use a `noqa` comment instead of a `skip_ansible_lint`.

Perhaps `noqa` should be listed as the preferred method in the docs since it doesn't have this limitation.

**Comment by gardar** [comment]
Another thought that came to mind, it would be good to get a info level log message when a skip is applied, as it can be pretty frustrating to get unexpected skips like this.
But that's perhaps a subject for another topic.
